services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/resource_tracker_core.py

```diff
@@ -0,0 +1,135 @@
+import datetime
+import json
+import logging
+from typing import Any
+
+import arrow
+from fastapi import FastAPI
+from models_library.users import UserID
+from prometheus_api_client import PrometheusConnect
+from simcore_service_resource_usage_tracker.modules.prometheus import (
+    get_prometheus_api_client,
+)
+
+# This script assumes everywhere that the minimum granularity of the data is 1 minute
+# Setting it smaller is unreasonable at least for prometheus, as the scraping interval is apprx. eq. to 1 h
+
+
+_logger = logging.getLogger(__name__)
+
+
+def _assure_dict_entry_exists(
+    metric_data, max_values_per_docker_id, image, userid
```

Comment:
missing types

Reply:
yes I should have written on the description but this module is the one @mrnicegyu11 is working on. so the rest kinds of wraps it. therefore this is going to change massively and is not currently "production ready"